Microgame/drinkcontest/basic functionality

WilliamDouris opened the pull request:
The basic functionality of the drink contest should work ! I try to make 3 difficulty balanced, if you can check that. I don't get the sprite from my artist, so I made ugly sprite to check if all goes right !
I don't know how (but I know) my code can't be more easier, if you have any improvement, DM me.
Like I said before I'm a little busy at the moment but I keep working on that ! 
Have a good day ! 
William 

PS : Whoops, I don't check if I built my microgame, on the reimusdodge !

WilliamDouris committed:
Drinking Party - Adding scene and traits

WilliamDouris committed:
Add Some temporary texture + Script detect Space

WilliamDouris committed:
Add Space input + Change size with space

WilliamDouris committed:
Input Percent_bottle + SetBottleBar (Scale + Pos)

WilliamDouris committed:
Add Control Animation + Victory end

WilliamDouris committed:
IA (use percent-click)

WilliamDouris committed:
3 level + sound

Gman8r reviewed (changes requested):
The microgame ID is DrinkContest, not DrinkingParty. You'll need to rename all the files:
- Rename the microgame folder to the correct name
- Highlight over every non-script asset and every folder in Unity with "AkiPotatoes" and press F2 to change it to have "DrinkContest"
- Then for each script, go into visual studio, move over the name "Public Class [Name]" and press F2 to rename that in Visual studio, and THEN rename the script itself in Unity with F2
- Commit and push to your pull request branch and it'll update it.
ALSO the Animations and Prefabs folders are empty. That's not good because Github won't save empty folders and Unity auto-removes their .metas, confusing our repo. Delete the empty folders (or populate them) and commit/push.

Other notes that you don't have to fix yet but keep in mind for future pull requests:
- The game is 8 beats, a mashing game shouldn't be long
- The third difficulty gets pretty impossible once you up the speed a few times. Pressing F during gameplay increases speed. It should still generally be doable up to speed 5 or 6 at stage 3.
- Your variable cases are pretty inconsistent. In general we use camelCase (firstWordIsLowercase) and not underscores, but either way your code can't seem to decide between caps and underscores for separating words.
- Using GetComponent<> every frame is generally slow. Use GetComponent<> once in Start()and store it in a variable to use for later.

WilliamDouris committed:
Fix for PR (description)

- Microgame Folder correct name
- Every non-script asset got DrinkContest+X name
- Script rename DrinkContest
- Folder empty are remove
---------------------
- The game is now 8 beats
- The difficulty is hard but possible in speed 6 stage 3.
- All variable are now under the form WordLikeThis
- The GetComponent<> is use only when I need to change the sprite (3 time by game).
------------------------------
+ The game still work !
